server/test/support/ProgramBuilder.java

```diff
@@ -296,6 +300,18 @@ public ProgramBuilder withProgramType(ProgramType programType) {
     return this;
   }
 
+  public ProgramBuilder withNotificationPreference(
```

Comment:
**suggestion (non-blocking):** Remove this method and always use setNotificationPreferences? Having two options seems unnecessarily complicated and this one is only used in tests, where I think it would actually be clearer to set all prefs at once. And no other fields in this class have two options afaict.